src/fparser/common/utils.py

```diff
@@ -380,3 +380,6 @@ def __new__(metacls, name, bases, dict):
         cls = type.__new__(metacls, name, bases, dict)
         _classes_cache[name] = cls
         return cls
+
+    def __getnewargs__(self):
+        return (self.name, self.bases, self.dict)
```

Comment:
Although we don't normally require a docstring for dunder methods, since this one is pretty specialist I think we will have one here. Probably just a cut-n-paste of the official docs:
Method to dictate the values passed to the [__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) method upon unpickling. The method must return a pair (args, kwargs) where args is a tuple of positional arguments and kwargs a dictionary of named arguments for constructing the object. Those will be passed to the [__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) method upon unpickling.

Reply:
This will also need to be covered by tests.

Reply:
I removed the method `__getnewargs__` of the class "classes" which was not required for the deepcopy()
=> full coverage :-)